### ISSUE TITLE
ssl_calls.c compile errors on devel branch

### ISSUE THREAD

**Issue by dajester2013** [neutrinolabs/xrdp#1050]
Trying to compile to test out the Hyper-V integration for Fedora 27, but run into this error during `make`:

```
make[2]: Entering directory '/opt/xrdp/common'
  CC       file.lo
  CC       log.lo
  CC       ssl_calls.lo
ssl_calls.c: In function ‘get_dh2236’:
ssl_calls.c:532:7: error: dereferencing pointer to incomplete type ‘DH {aka struct dh_st}’
     dh->p = BN_bin2bn(dh2236_p, sizeof(dh2236_p), NULL);
       ^~
make[2]: *** [Makefile:523: ssl_calls.lo] Error 1
make[2]: Leaving directory '/opt/xrdp/common'
make[1]: *** [Makefile:482: all-recursive] Error 1
make[1]: Leaving directory '/opt/xrdp'
make: *** [Makefile:414: all] Error 2
```

**Comment by jsorg71** [comment]
what openssl version is this?
can you find struct dh_st in file:///usr/include/openssl/dh.h ?

**Comment by dajester2013** [comment]
@jsorg71 

```
> openssl version
OpenSSL 1.1.0g-fips 2 Nov 2017
```

From dh.h: 
```
/* Already defined in ossl_typ.h */
/* typedef struct dh_st DH; */
```

From ossl_typ.h:
```
typedef struct dh_st DH;
```

**Comment by metalefty** [comment]
It depends on your OpenSSL version. It seems OpenSSL 1.1.0 causes compile error.
@speidy is working on the issue at #1063.

**Comment by metalefty** [comment]
Should be fixed by #1063.
